h/views/activity.py

```diff
@@ -205,6 +205,19 @@ def user_annotation_count(aggregation, userid):
 
         return result
 
+    def _get_total_annotations_in_group(self, result, q, request):
+        """
+        Get number of annotations in group.
+
+        If the search result already has this number don't run a query, just re-use it.
+        """
+        group_annotation_count = result['search_results'].total
+        if len(q) > 1:
```

Comment:
This is a good optimization to use, although I think the test for whether it can be safely used is perhaps a little brittle. A more explicit test for whether the search result is for a single group would be preferable.